File: Album/src/main/java/gorden/album/entity/Picture.java
package gorden.album.entity;

/**
 * 照片实体类
 * Created by Gordn on 2017/4/1.
 */

public class Picture {
    public String id;
    public String name;       //图片的名字
    public String path;       //图片的路径
    public long size;         //图片的大小
    public int width;         //图片的宽度
    public int height;        //图片的高度
    public String mimeType;   //图片的类型
    public long addTime;      //图片的创建时间

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;

        Picture picture = (Picture) o;

        return id != null ? id.equals(picture.id) : picture.id == null;

    }

    @Override
    public int hashCode() {
        return id != null ? id.hashCode() : 0;
    }
}

File: lib_album/src/main/kotlin/gorden/library/entity/Picture.kt
package gorden.library.entity

/**
 * 照片实体
 */
data class Picture(val id:String,
                   val name:String,
                   val path:String,
                   val size:Long,
                   val mineType:String,
                   val addTime:Long) {

    override fun hashCode(): Int {
        return id.hashCode()
    }

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (javaClass != other?.javaClass) return false

        other as Picture

        if (id != other.id) return false

        return true
    }

}